File: apkEditor/src/main/java/com/mcal/apkeditor/editor/TextEditor.java
package com.mcal.apkeditor.editor;

import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import androidx.annotation.NonNull;

import com.mcal.apkeditor.R;
import com.mcal.apkeditor.activities.SettingEditorActivity;
import com.mcal.apkeditor.activities.TextEditBigActivity;
import com.mcal.apkeditor.activities.TextEditNormalActivity;
import com.mcal.common.utils.ActivityUtils;

import java.io.File;
import java.util.ArrayList;

public class TextEditor {
    private static boolean isBigFile(Context ctx, String filepath) {
        boolean isBigFile = false;
        File f = new File(filepath);
        if (f.exists() && f.length() > SettingEditorActivity.getBigFileThreshold(ctx) * 1024L) {
            isBigFile = true;
        }
        if (isBigFile) {
            Toast.makeText(ctx, R.string.use_bfe_tip, Toast.LENGTH_SHORT).show();
        }
        return isBigFile;
    }


    @NonNull
    public static Intent getEditorIntent(Context ctx, String filepath, String apkPath) {
        Intent intent;
        if (isBigFile(ctx, filepath)) {
            intent = new Intent(ctx, TextEditBigActivity.class);
        } else {
            intent = new Intent(ctx, TextEditNormalActivity.class);
        }
        ActivityUtils.attachParam(intent, "xmlPath", filepath);
        if (apkPath != null) {
            ActivityUtils.attachParam(intent, "apkPath", apkPath);
        }
        return intent;
    }

    @NonNull
    public static Intent getEditorIntent(Context ctx, @NonNull ArrayList<String> filePathList, int index, String apkPath) {
        Intent intent;
        if (isBigFile(ctx, filePathList.get(index))) {
            intent = new Intent(ctx, TextEditBigActivity.class);
        } else {
            intent = new Intent(ctx, TextEditNormalActivity.class);
        }
        ActivityUtils.attachParam(intent, "fileList", filePathList);
        ActivityUtils.attachParam(intent, "curFileIndex", index);
        if (apkPath != null) {
            ActivityUtils.attachParam(intent, "apkPath", apkPath);
        }
        return intent;
    }
}

File: codeeditor/src/main/java/com/mcal/editor/TextEditor.kt
package com.mcal.editor

import android.content.Context
import android.content.Intent
import com.mcal.common.utils.ActivityHelper

object TextEditor {
    @JvmStatic
    fun getSoraEditor(context: Context,
                      filepath: String,
                      apkPath: String?,
                      startLine: Int,
                      searchString: String?
    ): Intent {
        val intent = Intent(context, EditorActivity::class.java)
        ActivityHelper.attachParam(intent, "filePath", filepath)
        apkPath?.let { ActivityHelper.attachParam(intent, "apkPath", apkPath) }
        ActivityHelper.attachParam(intent, "startLine", startLine)
        searchString?.let { ActivityHelper.attachParam(intent, "searchString", searchString) } // todo
        // extraString?.let { ActivityUtils.attachParam(intent, "extraString", extraString) } // todo delete
        return intent
    }

    @JvmStatic
    fun getSoraEditor(
        context: Context,
        filePathList: ArrayList<String?>?,
        index: Int,
        apkPath: String?,
        startLineList: ArrayList<Int>?,
        searchString: String?
    ): Intent {
        val intent = Intent(context, EditorActivity::class.java)
        filePathList?.let { intent.putStringArrayListExtra("fileList", filePathList) }
        intent.putExtra("curFileIndex", index)
        apkPath?.let { intent.putExtra("apkPath", apkPath) }
        startLineList?.let { intent.putIntegerArrayListExtra("startLineList", startLineList) }
        searchString?.let { intent.putExtra("searchString", searchString) } // todo
        //Toast.makeText(context, startLineList!![index].toString(), Toast.LENGTH_LONG).show()
        return intent
    }

    @JvmStatic
    fun getSoraEditor(
        context: Context,
        tmpFilePath: String,
        filePath: String,
        isRootMode: Boolean,
        ints: IntArray?
    ): Intent {
        val intent = Intent(context, EditorActivity::class.java)
        ActivityHelper.attachParam(intent, "filePath", tmpFilePath)
        ActivityHelper.attachParam(intent, "realFilePath", filePath)
        ActivityHelper.attachParam(intent, "isRootMode", isRootMode)
        ints?.let { ActivityHelper.attachParam(intent, "resourceIds", ints) }
        return intent
    }
}